### PR TITLE
Feat: Changelly

### PR DESCRIPTION
**Pull request by Gamboster**

### CODE REVIEW

File: packages/bitcore-wallet-client/src/lib/api.ts [hunk 1]
```diff
@@ -3089,4 +3089,43 @@ export class API extends EventEmitter {
       );
     });
   }
+
+  changellyGetPairsParams(data): Promise<any> {
```
Comment (matiu) [inline packages/bitcore-wallet-client/src/lib/api.ts:3093]: not on this PR, but we should separate integrations with core functions.

core.ts
integrations/changelly.ts
integrations/other.ts

Comment (matiu) [reply packages/bitcore-wallet-client/src/lib/api.ts:3093]: also in BWS's server.ts.

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 2]
```diff
@@ -4485,6 +4485,259 @@ export class WalletService {
     });
   }
 
+  changellyGetKeys(req) {
+    if (!config.changelly) throw new Error('Changelly missing credentials');
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/server.ts:4489]: maybe better: "ClientError: Service not configured."
and log.warn: "Changelly missing credentials"

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 3]
```diff
@@ -4485,6 +4485,259 @@ export class WalletService {
...
+      this.request.post(
+        URL,
+        {
+          headers,
+          body: message,
+          json: true
+        },
+        (err, data) => {
+          if (err) {
+            return reject(err.body ? err.body : err);
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/server.ts:4639]: `err?.body ?? err`

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 4]
```diff
@@ -4485,6 +4485,259 @@ export class WalletService {
...
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/server.ts:4688]: see above.

File: packages/bitcore-wallet-service/test/integration/server.js [hunk 5]
```diff
@@ -9805,6 +9805,347 @@ describe('Wallet service', function() {
     });
   });
 
+  describe('Changelly', () => {
```
Comment (matiu) [inline packages/bitcore-wallet-service/test/integration/server.js:9808]: move to a new file: test/integration/changelly.js

File: packages/bitcore-wallet-service/src/lib/server.ts (hunk 2 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/server.ts]: Done

File: packages/bitcore-wallet-service/src/lib/server.ts (hunk 3 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/server.ts]: Done

File: packages/bitcore-wallet-service/src/lib/server.ts (hunk 4 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/server.ts]: Done

File: packages/bitcore-wallet-service/test/integration/server.js (hunk 5 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/test/integration/server.js]: Done

**Review by matiu — APPROVED** [review]

**Comment by lorr756** [comment]
Hey eupepsia op de lucoció de pas psYra I'll oxaa as ⚡️🦐🏢🦐🌘🌘🌘🌘🌘🦋🦟🦐🦐🦟🌻⚡️🦐Sddsz.Z🏢🦓👩🏻‍🚒☹️☹️🙁

**Comment by lorr756** [comment]
Zhdpys was lass es I'll r out qaatuxtied cis g ur ir

**Pull request merged**